Link errors with suitesparse_config

I tried building dealii-8.5.0 with a suitesparse-4.4.3 (with static libraries) and encountered "undefined reference to SuiteSparse_malloc..." while building.  I noticed the suitesparseconfig library, which defines those symbols, was not included on the link, and the `LAPACK_SUITESPARSECONFIG_WITH_PIC` check determines whether or not that library is including in linking.  

I then upgraded my suitesparse to version 4.5.5, which installs shared libraries by default and should be build with PIC.  I reconfigured and got the same errors.

Now I noticed that the `LAPACK_SUITESPARSECONFIG_WITH_PIC` cmake check was still failing.  I would have expected it to succeed since my suitesparse-4.5.5 built every object with `-fPIC`.

I tried recreating the cmake test in my own sandbox:
```
$ cat >conftest.cxx
extern int SuiteSparse_version (int[3]);
  void foo(int bar[3]) { SuiteSparse_version(bar);}
$ c++ -c conftest.cxx
$ c++ -shared -o conftest.so conftest.o /path/to/suitesparse/lib/libsuitesparse.so
ld: conftest.o: relocation R_X86_64_PC32 against undefined symbol `_Z19SuiteSparse_versionPi' can not be used when making a shared object; recompile with -fPIC
ld: final link failed: Bad value
collect2: error: ld returned 1 exit status
```
which is the error cmake failed `LAPACK_SUITESPARSECONFIG_WITH_PIC`  for.  Out of curiosity I recompiled `conftest.cxx` with `-fPIC`:
```
$ c++ -fPIC -c conftest.cxx
$ c++ -shared -o conftest.so conftest.o /path/to/suitesparse/lib/libsuitesparse.so
$ echo $?
0
```
So that worked.  The `-fPIC` was absent from cmake's check.

As a workaround, I configured with `-DCMAKE_POSITION_INDEPENDENT_CODE:BOOL=ON`, and the `LAPACK_SUITESPARSECONFIG_WITH_PIC` test now passes and the build continues.

--- a/cmake/modules/FindUMFPACK.cmake
+++ b/cmake/modules/FindUMFPACK.cmake
@@ -119,24 +119,6 @@ FIND_UMFPACK_LIBRARY(CAMD camd)
 FIND_UMFPACK_LIBRARY(SuiteSparse_config suitesparseconfig)
 
 #
-# Test whether libsuitesparseconfig.xxx can be used for shared library
-# linkage. If not, exclude it from the command line.
-#
-LIST(APPEND CMAKE_REQUIRED_LIBRARIES
-  "-shared"
-  ${SuiteSparse_config_LIBRARY}
-  )
-CHECK_CXX_SOURCE_COMPILES("extern int SuiteSparse_version (int[3]);
-  void foo(int bar[3]) { SuiteSparse_version(bar);}"
-  LAPACK_SUITESPARSECONFIG_WITH_PIC
-  )
-RESET_CMAKE_REQUIRED()
-
-IF(LAPACK_SUITESPARSECONFIG_WITH_PIC OR NOT BUILD_SHARED_LIBS)
-  SET(_suitesparse_config SuiteSparse_config_LIBRARY)
-ENDIF()
-
-#
 # Add rt to the link interface as well (for whatever reason,
 # libsuitesparse.so depends on clock_gettime but the shared
 # lib does not record its dependence on librt.so as evidenced
@@ -148,7 +130,7 @@ MARK_AS_ADVANCED(rt_LIBRARY)
 DEAL_II_PACKAGE_HANDLE(UMFPACK
   LIBRARIES
     REQUIRED UMFPACK_LIBRARY
-    OPTIONAL CHOLMOD_LIBRARY CCOLAMD_LIBRARY COLAMD_LIBRARY CAMD_LIBRARY ${_suitesparse_config}
+    OPTIONAL CHOLMOD_LIBRARY CCOLAMD_LIBRARY COLAMD_LIBRARY CAMD_LIBRARY SuiteSparse_config_LIBRARY
     REQUIRED AMD_LIBRARY
     OPTIONAL METIS_LIBRARIES LAPACK_LIBRARIES rt_LIBRARY
   INCLUDE_DIRS

